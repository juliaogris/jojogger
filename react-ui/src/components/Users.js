import React, { Component, PropTypes } from 'react'

export default class Users extends Component {
  constructor () {
    super()
    this.state = {
      users: []
    }
  }

  render () {
    return (
      <div>Users</div>
    )
  }
}

Users.propTypes = {
  authedUser: PropTypes.object.isRequired }
