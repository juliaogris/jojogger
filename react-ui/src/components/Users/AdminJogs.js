import React, { Component } from 'react'
import PropTypes from 'prop-types'
import 'babel-polyfill'
import moment from 'moment'
import ErrorMessage from '../elements/ErrorMessage'
import JogsList from '../Jogs/JogsList'
import JogForm from '../Jogs/JogForm'
import { createJog, updateJog, deleteJog, getJogs } from '../../util/api'

export default class AdminJogs extends Component {
  constructor () {
    super()
    this.state = {
      loading: true,
      view: 'list', // ['list', 'add', 'edit']
      editJog: null,
      error: null,
      jogs: [],
      jogsInRange: [],
      startDate: null,
      endDate: null
    }
    this.handleEditClick = this.handleEditClick.bind(this)
    this.handleAddClick = this.handleAddClick.bind(this)
    this.gotoList = this.gotoList.bind(this)
    this.handleUpdateJog = this.handleUpdateJog.bind(this)
    this.handleDeleteJog = this.handleDeleteJog.bind(this)
    this.handleCreateJog = this.handleCreateJog.bind(this)
    this.handleDatesChange = this.handleDatesChange.bind(this)
    this.fetchJogs = this.fetchJogs.bind(this)
  }

  componentWillMount () {
    this.mounted = true
    this.fetchJogs()
  }

  componentWillUnmount () {
    this.mounted = false
  }

  async fetchJogs () {
    const { user, admin } = this.props
    this.setState({ loading: true })
    const jogs = await getJogs(user, admin)
    this.setState({ jogs, jogsInRange: jogs, loading: false, view: 'list', startDate: null, endDate: null })
  }

  handleDatesChange ({ startDate, endDate }) {
    const between = j => moment(j.date).isBetween(startDate, endDate, 'day', '[]')
    const jogsInRange = this.state.jogs.filter(between)
    this.setState({ startDate, endDate, jogsInRange })
  }

  handleEditClick (id) {
    const { jogs } = this.state
    const jog = jogs.find(jog => jog.id === id)
    const editJog = { ...jog }
    this.setState({ view: 'edit', editJog })
  }

  handleAddClick (event) {
    event.preventDefault()
    this.setState({ view: 'add' })
  }

  gotoList () {
    this.setState({ view: 'list' })
  }

  crudJog (jog, apiFunc) {
    this.setState({ loading: true })
    const { user, admin } = this.props
    apiFunc(user, jog, admin)
      .then(j => this.fetchJogs())
      .catch(error => {
        if (this.mounted) {
          this.setState({ error, loading: false })
        }
      })
  }

  handleUpdateJog (jog) {
    this.crudJog(jog, updateJog)
  }

  handleDeleteJog (jog) {
    this.crudJog(jog, deleteJog)
  }

  handleCreateJog (jog) {
    this.crudJog(jog, createJog)
  }

  render () {
    const { view, editJog, loading, error, jogs, startDate, endDate, jogsInRange } = this.state
    const { user, onCancel } = this.props
    if (loading || this.props.loading) {
      return <h1>Loading</h1>
    }
    if (error) {
      return <ErrorMessage error={error} onCancel={() => { this.setState({ error: null }) }} />
    }
    if (view === 'add') {
      return <JogForm
        createJog={this.handleCreateJog}
        onCancel={this.gotoList}
        heading={`User ${user.email}`} />
    }
    if (jogs.length === 0) {
      return (
        <div className='page'>
          <button onClick={onCancel} className='cancel-button'>Ã—</button>
          <p className='info'>No jogs tracked for <strong>{user.email}</strong>. <br />
            <button onClick={() => { this.setState({ view: 'add' }) }}>
            Add one.
          </button>
          </p>
        </div>
      )
    }
    if (view === 'list') {
      return <JogsList
        onEdit={this.handleEditClick}
        onAddClick={this.handleAddClick}
        onDatesChange={this.handleDatesChange}
        jogs={jogsInRange}
        noJog={jogs.length === 0}
        heading={`User ${user.email}`}
        startDate={startDate}
        endDate={endDate}

        {...this.props} />
    }
    if (view === 'edit') {
      return <JogForm
        updateJog={this.handleUpdateJog}
        deleteJog={this.handleDeleteJog}
        onCancel={this.gotoList}
        jogs={jogs}
        jog={editJog}
        heading={`User ${user.email}`}
      />
    }
  }
}

AdminJogs.propTypes = {
  user: PropTypes.object.isRequired,
  admin: PropTypes.object.isRequired,
  onCancel: PropTypes.func.isRequired
}
