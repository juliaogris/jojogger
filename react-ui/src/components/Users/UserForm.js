import React, { Component, PropTypes } from 'react'

import TickButton from '../elements/TickButton'
import DeleteButton from '../elements/DeleteButton'
import { getEmailError, getPasswordError } from '../../util/funcs'

const RoleSelect = ({ authedRole, role, onChange }) => {
  if (authedRole !== 'admin') {
    return null
  }
  return (
    <label>
      Role
      <select value={role} onChange={onChange} name='role'>
        <option value='regular'>Regular</option>
        <option value='manager'>Manager</option>
        <option value='admin'>Admin</option>
      </select>
    </label>)
}

export default class UserForm extends Component {
  constructor (props) {
    super(props)
    const { user } = this.props
    this.state = {
      email: user ? user.email : '',
      password: '',
      role: user ? user.role : 'regular',
      emailError: null,
      passwordError: null
    }
    this.valiemailInput = this.valiemailInput.bind(this)
    this.handleInputChange = this.handleInputChange.bind(this)
    this.handleDelete = this.handleDelete.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
    this.handleCancel = this.handleCancel.bind(this)
  }

  handleInputChange (event) {
    const { name, value } = event.target
    this.setState({ [name]: value, [name + 'Error']: null })
  }

  valiemailInput () {
    const { email, password, role } = this.state
    const errors = {}
    const emailError = getEmailError(email)
    if (emailError) {
      errors.emailError = emailError.message
    }
    if (password) {
      const passwordError = getPasswordError(password)
      if (passwordError) {
        errors.passwordError = 'Password min 6 letters'
      }
    }

    if (Object.keys(errors).length !== 0) {
      this.setState(errors)
      return null
    }
    const newUser = { email, role }
    if (password) {
      newUser.password = password
    }
    const { user } = this.props
    if (user) {
      newUser.id = user.id
    }
    return newUser
  }

  handleSubmit (event) {
    event.preventDefault()
    const { updateUser, createUser } = this.props
    const newUser = this.valiemailInput()
    if (!newUser) {
      return
    }
    if (updateUser) {
      updateUser(newUser)
    } else if (createUser) {
      createUser(newUser)
    }
  }

  handleDelete (event) {
    event.preventDefault()
    const { deleteUser, user } = this.props
    if (!deleteUser || !user) {
      return
    }
    deleteUser(user)
  }

  handleCancel (event) {
    event.preventDefault()
    this.props.onCancel()
  }

  render () {
    const canDelete = !!this.props.user
    const { authedRole } = this.props
    const { email, password, role } = this.state
    const { emailError, passwordError } = this.state
    return (
      <form className='jojog-form' action={this.handleSubmit}>
        <label className={emailError ? 'label-error' : ''}>
          {emailError || 'Email'}
          <input
            type='email'
            name='email'
            value={email}
            onChange={this.handleInputChange}
            className={emailError ? 'input-error' : ''}
          />
        </label>
        <label className={passwordError ? 'label-error' : ''}>
          {passwordError || 'Password'}
          <input
            type='password'
            name='password'
            value={password}
            onChange={this.handleInputChange}
            className={passwordError ? 'input-error' : ''}
          />
        </label>
        <RoleSelect authedRole={authedRole} role={role} onChange={this.handleInputChange} />
        <DeleteButton show={canDelete} onClick={this.handleDelete} item='User' />
        <TickButton onClick={this.handleSubmit} />
        <button onClick={this.handleCancel} className='cancel-button'>Ã—</button>
      </form>
    )
  }
}

UserForm.PropTypes = {
  onCancel: PropTypes.func.isRequired,
  authedRole: PropTypes.func.isRequired
}
