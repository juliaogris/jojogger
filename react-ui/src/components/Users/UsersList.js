import React from 'react'
import PropTypes from 'prop-types'
import AddButton from '../elements/AddButton'
import EditButton from '../elements/EditButton'
import JogsButton from '../elements/JogsButton'
import ErrorMessage from '../elements/ErrorMessage'

const Row = ({ user, onEdit, onJogsClick, admin }) =>
  (<tr>
    <td>{user.email}</td>
    { admin && <td>{user.role}</td> }
    { admin && <td><JogsButton onJogsClick={onJogsClick} id={user.id} /></td> }
    <td> <EditButton onEdit={onEdit} id={user.id} /> </td>
  </tr>)

const UsersList = (props) => {
  const { users, onEdit, onAddClick, authedUser, onJogsClick } = props
  const admin = authedUser.role === 'admin'
  if (!users) {
    return null
  }
  let filteredUsers = users
  if (!admin) {
    filteredUsers = users.filter(u => u.role === 'regular')
  }
  if (filteredUsers.length === 0) {
    return <p className='info'>No users with "regular" role.</p>
  }
  return (
    <div className='page'>
      <ErrorMessage {...props} />
      <table>
        <thead>
          <tr>
            <td className='no-top'>Email</td>
            {admin && <td className='no-top'>Role</td> }
            {admin && <td className='no-top' /> }
            <td className='no-top'><AddButton onClick={onAddClick} /></td>
          </tr>
        </thead>
        <tbody>
          { filteredUsers.map(user =>
            <Row
              key={user.id}
              user={user}
              onEdit={onEdit}
              admin={admin}
              onJogsClick={onJogsClick}
            />) }
        </tbody>
      </table>
    </div>
  )
}

UsersList.PropTypes = {
  users: PropTypes.array.isRequired,
  onEdit: PropTypes.func.isRequired,
  onJogsClick: PropTypes.func.isRequired,
  onAddClick: PropTypes.func.isRequired
}
export default UsersList
