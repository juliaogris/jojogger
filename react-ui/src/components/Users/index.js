import React, { Component } from 'react'
import PropTypes from 'prop-types'
import ErrorMessage from '../elements/ErrorMessage'
import { getUsers, createUser, updateUser, deleteUser } from '../../util/api'
import AdminJogs from './AdminJogs'
import UsersList from './UsersList'
import UserForm from './UserForm'

export default class Users extends Component {
  constructor () {
    super()
    this.state = {
      view: 'list', // 'add', 'edit, jogs'
      editUser: null,
      error: null,
      loading: false
    }
    this.fetchUsers = this.fetchUsers.bind(this)
    this.handleEditClick = this.handleEditClick.bind(this)
    this.handleJogsClick = this.handleJogsClick.bind(this)
    this.handleAddClick = this.handleAddClick.bind(this)
    this.crudUser = this.crudUser.bind(this)
    this.gotoList = this.gotoList.bind(this)
    this.handleUpdateUser = this.handleUpdateUser.bind(this)
    this.handleDeleteUser = this.handleDeleteUser.bind(this)
    this.handleAddUser = this.handleAddUser.bind(this)
  }

  gotoList () {
    this.setState({ view: 'list' })
  }

  handleEditClick (id) {
    const { users } = this.props
    const user = users.find(u => u.id === id)
    const editUser = { ...user }
    this.setState({ view: 'edit', editUser })
  }

  handleJogsClick (id) {
    const { users } = this.props
    const user = users.find(u => u.id === id)
    this.setState({ view: 'jogs', editUser: user })
  }

  handleAddClick (event) {
    event.preventDefault()
    this.setState({ view: 'add' })
  }

  componentWillMount () {
    this.mounted = true
    const { users } = this.props
    if (!users || users.length === 0) {
      this.fetchUsers()
    }
  }

  componentWillUnmount () {
    this.mounted = false
  }

  async fetchUsers () {
    const { setUsers } = this.props
    getUsers(this.props.authedUser)
      .then((users) => setUsers(users))
      .catch((error) => this.mounted && this.setState({ error }))
  }

  crudUser (user, apiFunc) {
    const { authedUser } = this.props
    if (authedUser.role !== 'admin') {
      user = { ...user }
      delete user.role
    }
    this.setState({ loading: true })
    apiFunc(authedUser, user)
      .then((u) => {
        return this.fetchUsers()
      })
      .then((users) => {
        this.props.setUsers(users)
      })
      .catch(error => {
        if (this.mounted) {
          this.setState({ error, loading: false })
        }
      })
  }

  handleUpdateUser (user) {
    this.crudUser(user, updateUser)
  }

  handleDeleteUser (user) {
    this.crudUser(user, deleteUser)
  }

  handleAddUser (user) {
    this.crudUser(user, createUser)
  }

  render () {
    const { view, editUser, error, loading } = this.state
    const { users, authedUser } = this.props
    const authedRole = authedUser.role
    if (error) {
      return <ErrorMessage error={error} onCancel={() => { this.setState({ error: null }) }} />
    }
    if (!users || users.length === 0 || loading) {
      return <p className='info'>Getting users...</p>
    }
    if (view === 'list') {
      return <UsersList
        users={users}
        onEdit={this.handleEditClick}
        onAddClick={this.handleAddClick}
        onJogsClick={this.handleJogsClick}
        {...this.props}
      />
    }
    if (view === 'add') {
      return <UserForm createUser={this.handleAddUser} onCancel={this.gotoList} authedRole={authedRole} />
    }
    if (view === 'edit') {
      return <UserForm
        updateUser={this.handleUpdateUser}
        deleteUser={this.handleDeleteUser}
        onCancel={this.gotoList}
        user={editUser}
        authedRole={authedRole}
      />
    }
    if (view === 'jogs') {
      return <AdminJogs
        onCancel={this.gotoList}
        user={editUser}
        admin={authedUser} />
    }
  }
}

Users.propTypes = {
  authedUser: PropTypes.object.isRequired,
  setUsers: PropTypes.func.isRequired
}
