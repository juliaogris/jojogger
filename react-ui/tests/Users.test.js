/* global it expect describe */
import React from 'react'
import ReactDOM from 'react-dom'
import Users from '../src/components/Users'
import UserForm from '../src/components/Users/UserForm'
import renderer from 'react-test-renderer'

const sampleUsers = [
  { 'email': 'admin@x.co', 'role': 'admin', 'id': 'id1' },
  { 'email': 'j@x.co', 'role': 'admin', 'id': 'id2' },
  { 'email': 'julia.ogris@gmail.com', 'role': 'regular', 'id': 'id3' },
  { 'email': 'manager@x.co', 'role': 'manager', 'id': 'id4' },
  { 'email': 'meetsshah@gmail.com', 'role': 'regular', 'id': 'id5' },
  { 'email': 'regular@x.co', 'role': 'regular', 'id': 'id6' }]

describe('Users', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Users
      setUsers={() => {}}
      authedUser={{}}
    />, div)
  })

  it('matches snapshot without ready users', () => {
    const tree = renderer.create(<Users
      setUsers={() => {}}
      authedUser={{}}
    />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot empty users', () => {
    const tree = renderer.create(<Users
      users={[]}
      setUsers={() => {}}
      authedUser={{}}
    />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot with users', () => {
    const tree = renderer.create(<Users
      setUsers={() => {}}
      authedUser={{}}
      users={sampleUsers}
    />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot with users for authed Admin', () => {
    const tree = renderer.create(<Users
      setUsers={() => {}}
      authedUser={{ role: 'admin' }}
      users={sampleUsers}
    />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('UserForm', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<UserForm
      onCancel={() => {}}
      authedRole='admin'
    />, div)
  })

  it('matches snapshot without user (add)', () => {
    const tree = renderer.create(<UserForm
      onCancel={() => {}}
      authedRole='admin'
    />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot with user (edit)', () => {
    const tree = renderer.create(<UserForm
      user={{ email: 'user@x.co', role: 'regular' }}
      onCancel={() => {}}
      authedRole='admin'
    />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot with user (edit) for Manager', () => {
    const tree = renderer.create(<UserForm
      user={{ email: 'user@x.co', role: 'regular' }}
      onCancel={() => {}}
      authedRole='manager'
    />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
