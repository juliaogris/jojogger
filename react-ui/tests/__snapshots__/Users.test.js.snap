exports[`UserForm matches snapshot with user (edit) 1`] = `
<form
  action={[Function]}
  className="jojog-form">
  <label
    className="">
    Email
    <input
      className=""
      name="email"
      onChange={[Function]}
      type="email"
      value="user@x.co" />
  </label>
  <label
    className="">
    Password
    <input
      className=""
      name="password"
      onChange={[Function]}
      type="password"
      value="" />
  </label>
  <label>
    Role
    <select
      name="role"
      onChange={[Function]}
      value="regular">
      <option
        value="regular">
        Regular
      </option>
      <option
        value="manager">
        Manager
      </option>
      <option
        value="admin">
        Admin
      </option>
    </select>
  </label>
  <div
    className="delete-row">
    <span
      className="delete-button"
      onClick={[Function]}>
      Delete this 
      User
    </span>
  </div>
  <div
    className="tick-button-wrap">
    <button
      className="tick-button"
      onClick={[Function]}
      type="submit">
      <svg
        className="add-icon"
        height="100%"
        viewBox="0 -1  41 42"
        width="100%"
        xmlns="http://www.w3.org/2000/svg">
        <g
          fill="none"
          fillRule="evenodd"
          stroke="none"
          strokeWidth="1">
          <polygon
            fill="white"
            fillRule="nonzero"
            points="14.5299498 13.3700502 15.9299498 11.9700502 26.9458691 22.4458691 20.9458691 28.4458691 19.5458691 27.0458691 24.1458691 22.4458691"
            transform="translate(20.737909, 20.207960) rotate(90.000000) translate(-20.737909, -20.207960) " />
        </g>
      </svg>
    </button>
  </div>
  <button
    className="cancel-button"
    onClick={[Function]}>
    ×
  </button>
</form>
`;

exports[`UserForm matches snapshot with user (edit) for Manager 1`] = `
<form
  action={[Function]}
  className="jojog-form">
  <label
    className="">
    Email
    <input
      className=""
      name="email"
      onChange={[Function]}
      type="email"
      value="user@x.co" />
  </label>
  <label
    className="">
    Password
    <input
      className=""
      name="password"
      onChange={[Function]}
      type="password"
      value="" />
  </label>
  <div
    className="delete-row">
    <span
      className="delete-button"
      onClick={[Function]}>
      Delete this 
      User
    </span>
  </div>
  <div
    className="tick-button-wrap">
    <button
      className="tick-button"
      onClick={[Function]}
      type="submit">
      <svg
        className="add-icon"
        height="100%"
        viewBox="0 -1  41 42"
        width="100%"
        xmlns="http://www.w3.org/2000/svg">
        <g
          fill="none"
          fillRule="evenodd"
          stroke="none"
          strokeWidth="1">
          <polygon
            fill="white"
            fillRule="nonzero"
            points="14.5299498 13.3700502 15.9299498 11.9700502 26.9458691 22.4458691 20.9458691 28.4458691 19.5458691 27.0458691 24.1458691 22.4458691"
            transform="translate(20.737909, 20.207960) rotate(90.000000) translate(-20.737909, -20.207960) " />
        </g>
      </svg>
    </button>
  </div>
  <button
    className="cancel-button"
    onClick={[Function]}>
    ×
  </button>
</form>
`;

exports[`UserForm matches snapshot without user (add) 1`] = `
<form
  action={[Function]}
  className="jojog-form">
  <label
    className="">
    Email
    <input
      className=""
      name="email"
      onChange={[Function]}
      type="email"
      value="" />
  </label>
  <label
    className="">
    Password
    <input
      className=""
      name="password"
      onChange={[Function]}
      type="password"
      value="" />
  </label>
  <label>
    Role
    <select
      name="role"
      onChange={[Function]}
      value="regular">
      <option
        value="regular">
        Regular
      </option>
      <option
        value="manager">
        Manager
      </option>
      <option
        value="admin">
        Admin
      </option>
    </select>
  </label>
  <div
    className="tick-button-wrap">
    <button
      className="tick-button"
      onClick={[Function]}
      type="submit">
      <svg
        className="add-icon"
        height="100%"
        viewBox="0 -1  41 42"
        width="100%"
        xmlns="http://www.w3.org/2000/svg">
        <g
          fill="none"
          fillRule="evenodd"
          stroke="none"
          strokeWidth="1">
          <polygon
            fill="white"
            fillRule="nonzero"
            points="14.5299498 13.3700502 15.9299498 11.9700502 26.9458691 22.4458691 20.9458691 28.4458691 19.5458691 27.0458691 24.1458691 22.4458691"
            transform="translate(20.737909, 20.207960) rotate(90.000000) translate(-20.737909, -20.207960) " />
        </g>
      </svg>
    </button>
  </div>
  <button
    className="cancel-button"
    onClick={[Function]}>
    ×
  </button>
</form>
`;

exports[`Users matches snapshot empty users 1`] = `
<p
  className="info">
  Getting users...
</p>
`;

exports[`Users matches snapshot with users 1`] = `
<div
  className="page">
  <table>
    <thead>
      <tr>
        <td
          className="no-top">
          Email
        </td>
        <td
          className="no-top">
          <div
            className="add-button-wrap">
            <button
              className="add-button"
              onClick={[Function]}>
              <svg
                className="add-icon"
                height="36px"
                viewBox="0 0 500 600"
                width="30px"
                xmlns="http://www.w3.org/2000/svg"
                xmlnsXlink="http://www.w3.org/1999/xlink">
                <path
                  d="M 241.99219,407.91016 L 241.99219,310.44922 L 144.72656,310.44922 L 144.72656,294.43359 L 241.99219,294.43359 L 241.99219,197.36328 L 257.61719,197.36328 L 257.61719,294.43359 L 355.27344,294.43359 L 355.27344,310.44922 L 257.61719,310.44922 L 257.61719,407.91016 L 241.99219,407.91016 z"
                  fill="white" />
              </svg>
            </button>
          </div>
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          julia.ogris@gmail.com
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
      <tr>
        <td>
          meetsshah@gmail.com
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
      <tr>
        <td>
          regular@x.co
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Users matches snapshot with users for authed Admin 1`] = `
<div
  className="page">
  <table>
    <thead>
      <tr>
        <td
          className="no-top">
          Email
        </td>
        <td
          className="no-top">
          Role
        </td>
        <td
          className="no-top" />
        <td
          className="no-top">
          <div
            className="add-button-wrap">
            <button
              className="add-button"
              onClick={[Function]}>
              <svg
                className="add-icon"
                height="36px"
                viewBox="0 0 500 600"
                width="30px"
                xmlns="http://www.w3.org/2000/svg"
                xmlnsXlink="http://www.w3.org/1999/xlink">
                <path
                  d="M 241.99219,407.91016 L 241.99219,310.44922 L 144.72656,310.44922 L 144.72656,294.43359 L 241.99219,294.43359 L 241.99219,197.36328 L 257.61719,197.36328 L 257.61719,294.43359 L 355.27344,294.43359 L 355.27344,310.44922 L 257.61719,310.44922 L 257.61719,407.91016 L 241.99219,407.91016 z"
                  fill="white" />
              </svg>
            </button>
          </div>
        </td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          admin@x.co
        </td>
        <td>
          admin
        </td>
        <td>
          <button
            className="jogs-button"
            onClick={[Function]}>
            Jogs
          </button>
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
      <tr>
        <td>
          j@x.co
        </td>
        <td>
          admin
        </td>
        <td>
          <button
            className="jogs-button"
            onClick={[Function]}>
            Jogs
          </button>
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
      <tr>
        <td>
          julia.ogris@gmail.com
        </td>
        <td>
          regular
        </td>
        <td>
          <button
            className="jogs-button"
            onClick={[Function]}>
            Jogs
          </button>
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
      <tr>
        <td>
          manager@x.co
        </td>
        <td>
          manager
        </td>
        <td>
          <button
            className="jogs-button"
            onClick={[Function]}>
            Jogs
          </button>
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
      <tr>
        <td>
          meetsshah@gmail.com
        </td>
        <td>
          regular
        </td>
        <td>
          <button
            className="jogs-button"
            onClick={[Function]}>
            Jogs
          </button>
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
      <tr>
        <td>
          regular@x.co
        </td>
        <td>
          regular
        </td>
        <td>
          <button
            className="jogs-button"
            onClick={[Function]}>
            Jogs
          </button>
        </td>
        <td>
           
          <button
            className="edit-button"
            onClick={[Function]}>
            <svg
              className=""
              height="14px"
              viewBox="0 0 600 600"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
              xmlnsXlink="http://www.w3.org/1999/xlink">
              <g
                fill="none"
                fillRule="evenodd"
                stroke="#777">
                <path
                  d="m70.064 422.35 374.27-374.26 107.58 107.58-374.26 374.27-129.56 21.97z"
                  strokeWidth="30" />
                <path
                  d="m70.569 417.81 110.61 110.61"
                  strokeWidth="25" />
                <path
                  d="m491.47 108.37-366.69 366.68"
                  strokeWidth="25" />
                <path
                  d="m54.222 507.26 40.975 39.546"
                  strokeWidth="25" />
              </g>
            </svg>
          </button>
           
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Users matches snapshot without ready users 1`] = `
<p
  className="info">
  Getting users...
</p>
`;
